client/blocks/three-d-secure/use-payment-intents.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').EmitResponseProps} EmitResponseProps
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ */
+
+/**
+ * Opens the modal for PaymentIntent authorizations.
+ *
+ * @param {Object}           params                Params object.
+ * @param {Stripe}           params.stripe         The stripe object.
+ * @param {Object}           params.paymentDetails The payment details from the
+ *                                                 server after checkout processing.
+ * @param {string}           params.errorContext   Context where errors will be added.
+ * @param {string}           params.errorType      Type of error responses.
+ * @param {string}           params.successType    Type of success responses.
+ */
+const openIntentModal = ( {
+	stripe,
+	paymentDetails,
+	errorContext,
+	errorType,
+	successType,
+} ) => {
+	const checkoutResponse = { type: successType };
+	if (
+		! paymentDetails.setup_intent &&
+		! paymentDetails.payment_intent_secret
+	) {
+		return checkoutResponse;
```

Comment:
Since this is invoked on every checkout processing response from the server, the current implementation means that the return value here will prevent any other subscribers hooked into the `onCheckoutAfterProcessingWithSucess` event emitter  from executing. This could be problematic if a Stripe express payment method is active in a checkout instance but a shopper uses a different payment method to make a purchase that also has something similar to a 3DS flow.

The solution for now I think is to modify this to:

```suggestion
		return true;
```

This requires me taking another look at this particular event emitter because I think it should handle `successType` responses as non blocking for any other subscribers to process, but the above should still work for your needs and eliminate the risk of breaking other payment method processing if present.

Reply:
Fixed in 6a3680ccb08ddde185979a6da5757439955797cf. Thank you Darren!